Flann failed to build on cmake 3.9 because of FindHDF5.cmake related error

I am on vs2015 and cmake 3.9. I am unable to build flann successfully because of FindHDF5.cmake related error. I suspect it's similar to the problem related to #1545. @ousttrue could you please help me look into this. I am trying to test #1283 and flann is one of the prerequisite. Below is the error log

``````
CMake Warning (dev) at CMakeLists.txt:17 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "MSVC" will no longer be dereferenced when the policy
  is set to NEW.  Since the policy is not set the OLD behavior will be used.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at CMakeLists.txt:21 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "MSVC" will no longer be dereferenced when the policy
  is set to NEW.  Since the policy is not set the OLD behavior will be used.
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Error at C:/Program Files/CMake/share/cmake-3.9/Modules/FindHDF5.cmake:450 (get_target_property):
  get_target_property called with incorrect number of arguments
Call Stack (most recent call first):
  C:/dev/vcpkg/scripts/buildsystems/vcpkg.cmake:145 (_find_package)
  cmake/flann_utils.cmake:25 (find_package)
  CMakeLists.txt:76 (find_hdf5)


CMake Warning at CMakeLists.txt:117 (message):
  gtest library not found, some tests will not be run


CMake Warning (dev) at C:/Program Files/CMake/share/cmake-3.9/Modules/FindOpenMP.cmake:212 (if):
  Policy CMP0054 is not set: Only interpret if() arguments as variables or
  keywords when unquoted.  Run "cmake --help-policy CMP0054" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  Quoted variables like "MSVC" will no longer be dereferenced when the policy
  is set to NEW.  Since the policy is not set the OLD behavior will be used.
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.9/Modules/FindOpenMP.cmake:324 (_OPENMP_GET_FLAGS)
  C:/dev/vcpkg/scripts/buildsystems/vcpkg.cmake:145 (_find_package)
  CMakeLists.txt:122 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at C:/Program Files/CMake/share/cmake-3.9/Modules/FindOpenMP.cmake:277 (if):
  if given arguments:

    "TRUE"

  An argument named "TRUE" appears in a conditional statement.  Policy
  CMP0012 is not set: if() recognizes numbers and boolean constants.  Run
  "cmake --help-policy CMP0012" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.9/Modules/FindOpenMP.cmake:380 (_OPENMP_GET_SPEC_DATE)
  C:/dev/vcpkg/scripts/buildsystems/vcpkg.cmake:145 (_find_package)
  CMakeLists.txt:122 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at C:/Program Files/CMake/share/cmake-3.9/Modules/FindOpenMP.cmake:277 (if):
  if given arguments:

    "TRUE"

  An argument named "TRUE" appears in a conditional statement.  Policy
  CMP0012 is not set: if() recognizes numbers and boolean constants.  Run
  "cmake --help-policy CMP0012" for policy details.  Use the cmake_policy
  command to set the policy and suppress this warning.
Call Stack (most recent call first):
  C:/Program Files/CMake/share/cmake-3.9/Modules/FindOpenMP.cmake:380 (_OPENMP_GET_SPEC_DATE)
  C:/dev/vcpkg/scripts/buildsystems/vcpkg.cmake:145 (_find_package)
  CMakeLists.txt:122 (find_package)
This warning is for project developers.  Use -Wno-dev to suppress it.

CMake Warning (dev) at test/CMakeLists.txt:4 (add_custom_target):
  Policy CMP0037 is not set: Target names should not be reserved and should
  match a validity pattern.  Run "cmake --help-policy CMP0037" for policy
  details.  Use the cmake_policy command to set the policy and suppress this
  warning.

  The target name "test" is reserved or not valid for certain CMake features,
  such as generator expressions, and may result in undefined behavior.
This warning is for project developers.  Use -Wno-dev to suppress it.

``````

ok. I try to build ``vcpkg install flann`` and get same error.

error message

```
CMake Error at C:/Program Files/CMake/share/cmake-3.9/Modules/FindHDF5.cmake:450 (get_target_property):
  get_target_property called with incorrect number of arguments
```

is problem.

I comment out ``find_package(HDF5)`` of ``flann-1.9.1/cmake/flann_utils.cmake`` , then I successfully build flann.

Error for me too (only with CMake 3.9):
```
CMake Error at D:/Work/vcpkg/downloads/cmake-3.9.0-win32-x86/share/cmake-3.9/Modules/FindHDF5.cmake:450 (get_target_property):
  get_target_property called with incorrect number of arguments
Call Stack (most recent call first):
  D:/Work/vcpkg/scripts/buildsystems/vcpkg.cmake:145 (_find_package)
  cmake/flann_utils.cmake:25 (find_package)
  CMakeLists.txt:76 (find_hdf5)
```

--- a/ports/flann/CONTROL
+++ b/ports/flann/CONTROL
@@ -1,3 +1,3 @@
 Source: flann
-Version: 1.9.1-4
+Version: 1.9.1-5
 Description: Fast Library for Approximate Nearest Neighbors

--- a/ports/flann/portfile.cmake
+++ b/ports/flann/portfile.cmake
@@ -25,6 +25,7 @@ vcpkg_configure_cmake(
         -DBUILD_PYTHON_BINDINGS=OFF
         -DBUILD_MATLAB_BINDINGS=OFF
         -DCMAKE_DEBUG_POSTFIX=-gd
+        -DHDF5_NO_FIND_PACKAGE_CONFIG_FILE=ON
 )
 
 vcpkg_install_cmake()

